javalib/src/main/scala/java/util/LinkedList.scala

```diff
@@ -0,0 +1,260 @@
+    addLast(e)
+    true
+  }
+
+  override def remove(o: Any): Boolean = {
+    val sizeBefore = inner.size
+    inner -= Box(o.asInstanceOf[E])
+    sizeBefore!= inner.size
+  }
+
+  override def addAll(c: Collection[_ <: E]): Boolean = {
+    val iter = c.iterator
+    var changed = false
+    while (iter.hasNext()) {
+      addLast(iter.next())
+      changed = true
+    }
+    changed
+  }
+
+  override def clear(): Unit =
+    inner.clear()
+
+  override def get(index: Int): E =
+    if (index < 0 || index >= size()) throw new IndexOutOfBoundsException(index.toString)
+    else inner(index).inner
+
+  override def set(index: Int, element: E): E = {
+    add(index, element)
+    remove(index+1)
+  }
+
+  override def add(index: Int, element: E): Unit =
+    if (index < 0 || index > size()) throw new IndexOutOfBoundsException(index.toString)
+    else inner.insert(index, Box(element))
+
+  override def remove(index: Int): E =
+    if (index < 0 || index >= size()) throw new IndexOutOfBoundsException(index.toString)
+    else inner.remove(index).inner
+
+  override def indexOf(o: Any): Int =
+    inner.indexOf(Box(o.asInstanceOf[E]))
+
+  override def lastIndexOf(o: Any): Int =
+    inner.lastIndexOf(Box(o.asInstanceOf[E]))
+
+  def peek(): E =
+    peekFirst()
+
+  def element(): E =
+    getFirst()
+
+  def poll(): E =
+    pollFirst()
+
+  def remove(): E =
+    remove(0)
+
+  def offer(e: E): Boolean =
+    offerLast(e)
+
+  def offerFirst(e: E): Boolean = {
+    val sizeBefore = size()
+    addFirst(e)
+    size() > sizeBefore
+  }
+
+  def offerLast(e: E): Boolean = {
+    val sizeBefore = size()
+    addLast(e)
+    size() > sizeBefore
+  }
+
+  def peekFirst(): E =
+    if (inner.isEmpty) null.asInstanceOf[E]
+    else inner.head.inner
+
+  def peekLast(): E =
+    if (inner.isEmpty) null.asInstanceOf[E]
+    else inner.last.inner
+
+  def pollFirst(): E = {
+    val ret = peekFirst()
+    if (ret == null) null.asInstanceOf[E] else {
+      inner.trimStart(1)
+      ret
+    }
+  }
+
+  def pollLast(): E = {
+    val ret = peekLast()
+    if (ret == null) null.asInstanceOf[E] else {
+      inner.trimEnd(1)
+      ret
+    }
+  }
+
+  def push(e: E): Unit =
+    addFirst(e)
+
+  def pop(): E =
+    removeFirst()
+
+  def removeFirstOccurrence(o: Any): Boolean = {
+    val index = indexOf(o)
+    if (index < 0 || index >= size()) false else {
+      remove(index)
+      true
+    }
+  }
+
+  def removeLastOccurrence(o: Any): Boolean = {
+    val index = lastIndexOf(o)
+    if (index < 0 || index >= size()) false else {
+      remove(index)
+      true
+    }
+  }
+
+  override def listIterator(_index: Int): ListIterator[E] = {
+    new ListIterator[E] {
+
+      if (_index < 0 || _index >= self.size())
+        throw new IndexOutOfBoundsException(_index.toString)
+
+      private var last: Int = _index
+      private var index: Int = _index
+
+      private var performedAdd = true
+      private var performedRemove = true
+
```

Comment:
You have too many variables. You could keep track of the state of `performedAdd ` and `performedRemove ` using `last == -1`. Check the `RandomAccessListIterator` in the `AbstractList` class.